internal/terraform/node_resource_abstract_instance.go

```diff
@@ -657,6 +665,12 @@ func (n *NodeAbstractResourceInstance) refresh(ctx EvalContext, deposedKey state
 			},
 		})
 
+		// If we don't support deferrals, but the provider reports a deferral and does not
+		// emit any error level diagnostics, we should emit an error.
+		if resp.Deferred != nil && !deferralAllowed && !resp.Diagnostics.HasErrors() {
```

Comment:
Could you explain a bit more about the final check in this conditional, the one for no errors yet in the response diagnostics? 

Is it just because... we're gonna be erroring anyway, and so who knows, maybe that incoherent provider response will resolve itself anyway once you fix the more urgent problem?

Reply:
The tools to build providers with (framework or sdk) do the same check in their implementation because they can provider a more detailed explanation what might have gone wrong to the user / provider developer. But providers can also be developed without helpers so in case this erroneous condition exists and no diagnostics have been emitted we want to provide this error to make sure users see at least and at most one error.